test/instrumentations/amqplib/amqplib.test.ts

```diff
@@ -0,0 +1,145 @@
+
+const startRabbitMqContainer = async () => {
+  return await new GenericContainer('rabbitmq:latest')
+    .withExposedPorts(DEFAULT_RABBITMQ_PORT)
+    .withWaitStrategy(Wait.forLogMessage('Server startup complete'))
+    .withStartupTimeout(STARTUP_TIMEOUT)
+    .start();
+};
+
+describe.each(versionsToTest(INSTRUMENTATION_NAME, INSTRUMENTATION_NAME))(
+  `Instrumentation tests for the ${INSTRUMENTATION_NAME} package`,
+  function (versionToTest) {
+    let testApp: TestApp;
+    let rabbitmqContainer: StartedTestContainer;
+
+    beforeAll(async function () {
+      reinstallPackages(TEST_APP_DIR);
+      fs.mkdirSync(SPANS_DIR, { recursive: true });
+      installPackage(TEST_APP_DIR, INSTRUMENTATION_NAME, versionToTest);
+
+      /*
+       * Warm up container infra, download images, etc.
+       * This prevents spurious failures of early tests.
+       */
+      try {
+        rabbitmqContainer = await startRabbitMqContainer();
+      } finally {
+        if (rabbitmqContainer) {
+          rabbitmqContainer.stop();
+        }
+      }
+    }, STARTUP_TIMEOUT /* Long timeout, this might have to pull Docker images */);
+
+    beforeEach(async function () {
+      rabbitmqContainer = await startRabbitMqContainer();
+
+      const host = rabbitmqContainer.getHost();
+      const port = rabbitmqContainer.getMappedPort(DEFAULT_RABBITMQ_PORT);
+
+      console.info(`RabbitMQ container started on ${host}:${port}...`);
+    }, 15_000);
+
+    afterEach(async function () {
+      console.info('Killing test app...');
+      await testApp.kill();
+      await rabbitmqContainer.stop();
+    });
+
+    afterAll(function () {
+      uninstallPackage(TEST_APP_DIR, INSTRUMENTATION_NAME, versionToTest);
+    });
+
+    itTest(
+      {
+        testName: `amqp roundtrip: ${versionToTest}`,
+        packageName: INSTRUMENTATION_NAME,
+        version: versionToTest,
+        timeout: TEST_TIMEOUT,
+      },
+      async function () {
+        const exporterFile = `${SPANS_DIR}/amqp-roundtrip.amqplib@${versionToTest}.json`;
+
+        testApp = new TestApp(TEST_APP_DIR, INSTRUMENTATION_NAME, exporterFile, {
+          OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT: '4096',
+        });
+
+        const topic = 'test-topic-roundtrip';
+        const message = 'test-message-roundtrip';
+        const host = rabbitmqContainer.getHost();
+        const port = rabbitmqContainer.getMappedPort(DEFAULT_RABBITMQ_PORT);
+        await testApp.invokeGetPath(
+          `/invoke-amqp-producer?topic=${topic}&message=${message}&host=${host}&port=${port}`
+        );
+
+        let spans = await testApp.invokeGetPathAndRetrieveSpanDump(
+          `/invoke-amqp-consumer?topic=${topic}&message=${message}&host=${host}&port=${port}`
+        );
+
+        /*
+         * TODO: HORRIBLE WORKAROUND: The internal span we are looking for seems to be closed LATER than
+         * the Server span, so we must busy-poll.
+         */
+        while (spans.length < 4) {
```

Comment:
an idea - I see this "horrible workaround" everywhere.
WDYT to extract it to a single util function, and we will fix it once [if] it will be possible?
There's a lot of code duplication around here